### PR TITLE
Fix illegal subroutine calls.

### PR DESCRIPTION
**Pull request by DJDavies2**
Addresses https://github.com/ecmwf-ifs/ectrans/issues/13.

I think the code concerned is not standard conforming, it appears in four places.  It concerns passing array elements to F77-style arrays; this is okay in many situations but in some circumstances it isn't.  For example if the array being passed is an assumed shape dummy argument it is not okay.  It is to do with array contiguity; passing arrays via single elements in this way relies on the arrays being known to be contiguous which for an assumed shape dummy argument is not the case (for example for the Fortran 2003 FCD see 12.4.1.2 several paragraphs down).

Code builds on all the platforms I have.

### CODE REVIEW

**Comment by FussyDuck** [comment]
[CLA assistant check](http://claassistant.ecmwf.int:80/ecmwf-ifs/ectrans?pullRequest=14) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](http://claassistant.ecmwf.int:80/ecmwf-ifs/ectrans?pullRequest=14) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](http://claassistant.ecmwf.int:80/check/ecmwf-ifs/ectrans?pullRequest=14) it.</sub>

File: src/trans/internal/ftdir_mod.F90 [hunk 1]
```diff
@@ -86,7 +86,7 @@ SUBROUTINE FTDIR(PREEL,KFIELDS,KGL)
 
     IF( T%LUSEFFT992(KGL) )THEN
 
-      CALL FFT992(PREEL(1,IOFF),T%TRIGS(1,KGL),&
+      CALL FFT992(PREEL(1:,IOFF:),T%TRIGS(1,KGL),&
```
Comment (wdeconinck) [inline src/trans/internal/ftdir_mod.F90:89]: The first argument to FFT992 is expected to be a 1-dimensional array.
Would it then not be sufficient to pass `PREEL(:,IOFF)` ?

Comment (DJDavies2) [reply src/trans/internal/ftdir_mod.F90]: Yes you are correct.  I have pushed an update to the branch.

**Comment by DJDavies2** [comment]
Is this okay to merge?

**Comment by wdeconinck** [comment]
> Is this okay to merge?

Yes it is, as soon as the CLA is signed by all contributors.

**Comment by wdeconinck** [comment]
Thank you @DJDavies2 for this fix!

**Pull request merged**